extract_require_changes.py

```diff
@@ -0,0 +1,149 @@
+
+import os
+import pandas as pd
+from tabulate import tabulate
+
+
+BASE_DIR      = "/Users/mac/Desktop/RepairComp/results/smartbugs"
+OUTPUT_CSV    = os.path.join(BASE_DIR, "data_analysis", "patches_w_require.csv")
+OUTPUT_TABLE  = os.path.join(BASE_DIR, "data_analysis", "pretty_table2.txt")
+GITHUB_BASE   = (
+    "https://github.com/ASSERT-KTH/RepairComp/blob/main/results/smartbugs"
+)
+
+def clean_keys(df: pd.DataFrame) -> pd.DataFrame:
+    for col in ("Tool", "Category", "Patch"):
+        df[col] = (
+            df[col]
+            .astype(str)
+            .str.strip()
+            .str.lower()
+            .str.replace(r"\s+", " ", regex=True)
+        )
+    return df
+
+
+def is_real_require(line: str) -> bool:
+    stripped = line.strip()[1:].strip()  
+    # remove +/-, och whitespace
+    has_require = "require(" in stripped
+    full_comment = stripped.startswith("//") or stripped.startswith("/*")
+    return has_require and not full_comment
+
+def classify_require_type(line: str) -> str:
+    stripped = line.strip()
+    if "require(" in stripped:
+        lower = stripped.lower()
+        if any(x in lower for x in ["call.value", "send(", "transfer(", "success", "now", "block.timestamp"]):
+            return "runtime check"
+        if "msg.sender" in lower and "==" in lower and "owner" in lower:
+            return "modifier"
```

Comment:
Could you motivate further what applies in the modifier category? 

I'm considering two options, either is a modified invariant or it comes from a modifier (ref: [solidity docs](https://docs.soliditylang.org/en/latest/contracts.html#function-modifiers))